tests/test_flow.py

```diff
@@ -338,3 +340,54 @@ def mapf(x):
     assert event.body == 'original body'
     assert result.key == 'new key'
     assert result.body == 'new body'
+
+
+def test_batch():
+    controller = build_flow([
+        Source(),
+        Batch(4, 5),
```

Comment:
If you don't want timeout to be reached, set it very high. 5s is just low enough that it might happen under some strange circumstances (e.g. in a highly virtualized environment).